x-pack/plugins/security/server/lib/check_license.js

```diff
@@ -27,6 +27,7 @@ export function checkLicense(xPackInfo) {
   // assume worst-case and lock user at login screen.
   if (!xPackInfo.isAvailable()) {
     return {
+      authenticate: true,
```

Comment:
What are your thoughts on naming this `shouldAuthenticate`? I'm not opposed to leaving it as-is, but `shouldAuthenticate` would match the name that we are exposing on the security plugin, and I feel it would be more consistent with the rest of the flags we are presenting in this license check response.

Other options: `isAuthenticated` / `allowAuthentication` (but this feels similar to `allowLogin`)


edit: If I'm reading this correctly, the `authenticate` flag value mirrors that of `showLogin`. Do you think it makes sense to consolidate these flags?

Reply:
Switching everything to use `allowAuthentication` from the license check results seems reasonable to me. I'm torn on whether or not we should expose `server.plugins.security.allowAuthentication` though or we should continue using `server.plugins.security.shouldAuthenticate`.